### DISCUSSION TITLE
Proxy Error with Nginx and Https enabled

### DISCUSSION THREAD

**Discussion by TheGuardianLight** [firefly-iii/firefly-iii#9012] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

To keep things simple, I installed Firefly III on my server using Portainer. However, as with any Docker container, it's behind my reverse proxy: Nginx.

However, having configured a Let's Encrypt certificate for my domain name redirecting to Firefly's server (and therefore to nginx), the interface became completely buggy, with nothing loading or working correctly.

### Debug information

Debug information generated at 2024-06-30 19:21:30 for Firefly III version **v6.1.18**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.18 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.8 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1104, base <span>#</span>85</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Paris + Europe/Paris</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>fr_FR + equal</td></tr><tr><td>Trusted proxies</td><td>195.15.223.109</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-06-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-06-30 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>fr_FR</td></tr><tr><td>User locale</td><td>fr_FR</td></tr><tr><td>Locale(s) supported</td><td>fr_FR.utf8: :white_check_mark:<br>fr_FR.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 Edg/127.0.0.0</td></tr></tbody></table>

### Expected behaviour

That pages and software load correctly in both http and https.

### Steps to reproduce

1. Install Firefly with Portainer like in the documentation.
2. Having a domain name configured and a reverse proxy configured with Nginx.
3. Having an SSL certificate installed with Let's Encrypt's certbot.

**Comment by JC5** [comment]
This is something you'll have to configure in nginx :) 

https://docs.firefly-iii.org/references/faq/install/#how-do-i-set-tls-in-firefly-iii-or-the-data-importer

**Reply by TheGuardianLight** [reply]
Well I have this configuration :

```conf
server {

      server_name **redacted**;

      location / {
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Scheme $scheme;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_redirect off;
        proxy_pass http://localhost:8085/;
        client_max_body_size 64M;
        proxy_read_timeout 300s;
      }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/**redacted**/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/**redacted**/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}
server {
    if ($host = **redacted**) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


      listen 80;

      server_name **redacted**;
    return 404; # managed by Certbot
}
```

And actually it's show me this :

![image](https://github.com/firefly-iii/firefly-iii/assets/64707127/f8fd6180-81fe-4031-ae70-79af8bde60c1)

And there is the navigator console output : 

![image](https://github.com/firefly-iii/firefly-iii/assets/64707127/af7c8f26-6fcb-49cb-9a4c-bbed91bdf133)

**Comment by TheGuardianLight** [comment]
Oh wait, I think I found the solution :
Just needed to modify http -> https inside the stack.env at `APP_URL` and `VANITY_URL`

**Reply by JC5** [reply]
Those aren't used by Firefly III to write the urls though, curious.

**Reply by ChiruMori** [reply]
I encountered the same issue in version 6.1.22, but after modifying the `APP_URL`, the problem is still not solved, I use Plain Docker.

![image](https://github.com/user-attachments/assets/94da41e2-5339-43d9-9c4f-51d8138959cc)
![image](https://github.com/user-attachments/assets/af7f84d6-02d4-437f-a927-6bac9dbc8848)

My run.sh:

```sh
docker run -d \
-v /root/docker/fireflyiii/upload:/var/www/html/storage/upload \
--name firefly \
-p 19716:8080 \
-e APP_KEY=xxx \
-e DB_HOST=172.17.33.243 \
-e DB_PORT=6603 \
-e DB_CONNECTION=mysql \
-e DB_DATABASE=firefly \
-e DB_USERNAME=mori \
-e DB_PASSWORD=xxx\
-e DEFAULT_LANGUAGE=zh_CN \
-e TZ=Asia/Shanghai \
-e APP_URL=https://xxx \
fireflyiii/core:latest
```

My nginx.conf:

```
    server {
        listen      443 ssl http2;
        server_name f.xxx;
        ssl_certificate  /root/ssl/f.xxx.pem;
        ssl_certificate_key  /root/ssl/f.xxx.key;
        ssl_session_cache  shared:SSL:1m;
    
        location / {
            
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Scheme $scheme;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_redirect off;
            client_max_body_size 64M;
            proxy_read_timeout 300s;
            proxy_pass  http://172.17.33.243:19716;
        }
    }
```

**Reply by JC5** [reply]
Please also set environment variable `TRUSTED_PROXIES=*`.

**Reply by ChiruMori** [reply]
> Please also set environment variable `TRUSTED_PROXIES=*`.

Cool, it works!

**Reply by tonyvilla90** [reply]
perfecto amigo tiene razon en casaos TRUSTED_PROXIES=* es solo poner * al lado ya perfecto funciona
